facts/test_datasources.py

```diff
@@ -108,6 +108,28 @@ def test_switchtypes_tsv():
     assert result, err
 
 
+def test_switchconfigs_tsv():
+    """test switchconfigs"""
+
+    directory = "../switch-configuration/config/types/"
+    for filename in os.listdir(directory):
+        meta = {
+            "file": directory + filename,
+            "header": False,
+            "count": "1+",
+            "cols": [
+                ds.isvalidport,
```

Comment:
having `isvalidport` and `isvalidlink` is definitely a great start to testing these switch config types, nice work!